docs/custom_instrumentation.md

```diff
@@ -0,0 +1,171 @@
+# Custom Instrumentation
```

Comment:
These examples are also useful, but I think we need more plain english explanations and de-duping of similar use cases. Also, I think I was already assigned very similar documentation work for custom events and following up on instrumentation-level guides.